src/tests/unit/mock_data/mock_recipe_provider.py

```diff
@@ -5,20 +5,23 @@
 from enums import StorageDataTypes
 
 
-def mock_storage_recipe_provider(type: str, context: str) -> list[StorageRecipe]:
-    all_storage_recipes = defaultdict(list)
-    with open("src/tests/unit/mock_data/mock_storage_recipes/mock_storage_recipes.json") as f:
-        all_storage_recipes.update(json.load(f))
-    return [
-        StorageRecipe(
-            name=sr.get("name"),
-            storage_affinity=StorageDataTypes(sr.get("storageAffinity", "default")),
-            attributes={
-                a["name"]: StorageAttribute(
-                    name=a["name"], contained=a["contained"], storage_affinity=a.get("storageAffinity", "default")
-                )
-                for a in sr.get("attributes", [])
-            },
-        )
-        for sr in all_storage_recipes[type]
-    ]
+def mock_storage_recipe_provider_generator(path_to_mock_storage_recipes: str = None):
```

Comment:
Interesting. Maybe this should be a class, and the `path` parameter could be a data field? Might get around this function-inside-function situation, which I find a little confusing. Would love to discuss this further.

Reply:
recipe_provider would then resemble blueprint_provider, which makes sense